src/main/java/cn/nukkit/level/Level.java

```diff
@@ -3264,6 +3264,9 @@ public Position getSafeSpawn(Vector3 spawn) {
         }
 
         if (spawn != null) {
+            if(!server.isSafeSpawn()) {
+                return new Position(spawn.x,spawn.y,spawn.z);
+            }
```

Comment:
Plugins can use this method to find a safe position to teleport the player. Rather, this check should be done where the getSafeSpawn() method is used, not here

Reply:
Ah sure, i just looked at the source of the issue, and my debbugger was completely broken. will fix.

Reply:
Moved to calling side.